packages/zoltan2/core/src/input/Zoltan2_MatrixAdapter.hpp

```diff
@@ -175,11 +175,11 @@ template <typename User, typename UserCoord=User>
       \param colIds on return will point to the global column Ids for
          the non-zeros for each row.
    */
-  virtual void getCRSView(const offset_t *&offsets, const gno_t *&colIds) const
+  virtual void getCRSView(ArrayRCP<const offset_t> &offsets, ArrayRCP<const gno_t> &colIds) const
```

Comment:
This change is a non-backward-compatible interface change -- one that will likely be noticed by no one.  I am willing to accept it, but I am curious why it was needed?  Ditto for other methods in this adapter.

Reply:
Pointers were going out of scope when I got them to be just in time.  You'd get the pointer and then the thing the pointer pointed at would go out of scope and it would be invalid.